File: jar-shell/src/test/java/com/beancounter/shell/TestCsvImport.java
package com.beancounter.shell;


import static org.assertj.core.api.Assertions.assertThat;

import com.beancounter.common.model.Portfolio;
import com.beancounter.common.utils.PortfolioUtils;
import com.beancounter.shell.csv.CsvIngester;
import com.beancounter.shell.ingest.HttpWriter;
import com.beancounter.shell.ingest.IngestionRequest;
import java.util.List;
import lombok.SneakyThrows;
import org.junit.jupiter.api.Test;

public class TestCsvImport {

  @Test
  @SneakyThrows
  void importCsv() {
    CsvIngester csvIngester = new CsvIngester();
    Portfolio portfolio = PortfolioUtils.getPortfolio("TEST");
    portfolio.setId("test");
    csvIngester.prepare(
        IngestionRequest.builder()
            .portfolio(portfolio)
            .file("/trades.csv").build(),
        new HttpWriter());
    List<List<String>> results = csvIngester.getValues();
    assertThat(results).isNotEmpty().hasSize(5);
  }
}

File: jar-shell/src/test/java/com/beancounter/shell/TestCsvImport.kt
package com.beancounter.shell

import com.beancounter.client.ingest.FxTransactions
import com.beancounter.client.ingest.RowAdapter
import com.beancounter.client.services.TrnService
import com.beancounter.shell.csv.CsvIngester
import com.beancounter.shell.ingest.HttpWriter
import com.beancounter.shell.ingest.IngestionRequest
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito

class TestCsvImport {

    private var trnService: TrnService = Mockito.mock(TrnService::class.java)

    private var rowAdapter: RowAdapter = Mockito.mock(RowAdapter::class.java)

    private var fxTransactions: FxTransactions = Mockito.mock(FxTransactions::class.java)

    @Test
    fun importCsv() {
        val csvIngester = CsvIngester()
        csvIngester.prepare(
            IngestionRequest(file = "/trades.csv"),
            HttpWriter(trnService, rowAdapter, fxTransactions)
        )
        val results = csvIngester.values
        Assertions.assertThat(results).isNotEmpty.hasSize(5)
    }
}
